taxi/views.py

```diff
@@ -79,6 +81,19 @@ class CarDeleteView(LoginRequiredMixin, generic.DeleteView):
     success_url = reverse_lazy("taxi:car-list")
 
 
+class UserCarUpdateView(View):
+    def post(self, request: HttpRequest, pk: int) -> HttpResponseRedirect:
+        car = get_object_or_404(Car, pk=pk)
+        if "assign" in request.POST:
+            car.drivers.add(request.user)
+        elif "remove" in request.POST:
+            car.drivers.remove(request.user)
+        return HttpResponseRedirect(reverse_lazy(
+            "taxi:car-detail",
+            args=[car.id])
+        )
```

Comment:
try to simplify ur solution
no need to pass some params inside post request, u just need to check if user is already assigned to the car